code/game/objects/items/weapons/cloaking_device.dm

```diff
@@ -0,0 +1,176 @@
+/obj/item/weapon/cloaking_device
+	name = "cloaking device"
+	desc = "Use this to become invisible to the human eye."
+	icon = 'icons/obj/device.dmi'
+	icon_state = "shield0"
+	var/active = 0.0
+	flags = CONDUCT
+	item_state = "electronic"
+	throwforce = 10.0
+	throw_speed = 2
+	throw_range = 10
+	w_class = 2.0
+	origin_tech = list(TECH_MAGNET = 3, TECH_ILLEGAL = 4)
+
+	var/power_usage = 35000//A high powered cell allows 5 minutes of continuous usage
+	//Note it can be toggled on and off easily. You can make it last an hour if you only use it when
+	//people are nearby to see. Carry spare/better cells for extended cloaking.
+	var/obj/item/weapon/cell/cell = null
+	var/mob/living/owner = null
+	var/datum/modifier/cloaking_device/modifier = null
+
+/obj/item/weapon/cloaking_device/New()
+	..()
+	cloaking_devices += src
+	cell = new /obj/item/weapon/cell/high(src)
+
+/obj/item/weapon/cloaking_device/Destroy()
+	..()
+	cloaking_devices -= src
+
+
+/obj/item/weapon/cloaking_device/equipped(var/mob/user, var/slot)
+	..()
+	//Picked up or switched hands or worn
+	register_owner(user)
+
+
+/obj/item/weapon/cloaking_device/dropped(var/mob/user)
+	..()
+	spawn(1)//Things are dropped on the floor briefly when being put into containers or swapping hands
+	//Its dumb. This little hack works around it
+		var/mob/M = get_holding_mob()
+		if(!M)
```

Comment:
and i'm using the dropped proc because this is FOR when its dropped.

This code handles cases where the cloaking device is dropped on the floor or thrown, the register function will remove the stealth from the person who was holding it